waspc/src/Wasp/Util/IO.hs

```diff
@@ -74,3 +80,15 @@ deleteFileIfExists filePath = do
   let filePathStr = SP.toFilePath filePath
   exists <- SD.doesFileExist filePathStr
   when exists $ SD.removeFile filePathStr
+
+doesFileExist :: Path' r (File f) -> IO Bool
+doesFileExist = SD.doesFileExist . SP.toFilePath
+
+readFile :: Path' r (File f) -> IO String
+readFile = P.readFile . SP.toFilePath
+
+writeFile :: Path' r (File f) -> String -> IO ()
+writeFile = P.writeFile . SP.toFilePath
+
+removeFile :: Path' r (File f) -> IO ()
+removeFile = SD.removeFile . SP.toFilePath
```

Comment:
I knew you were going to mention this, but forgot to put in a comment. Nice catch!

Anyway, I had this dilemma. Here are my thoughts:

> It helps you avoid mistakes where you are trying to do something with the file, but you by accident provided rel path instead of abs and it is not relative to what you wanted it to be, and you are in a problem.

This is true, but you could still just use `toFilePath` (and that's exactly what was used in half of these places). In other words, we have no way of forcing the caller to properly perform the conversions (and this way at least we know it's not a `Dir`).

> One option would be to add more functions here: readFileAbs, removeFileAbs, ... .

I also thought about doing this, but decided against it in favor of matching the already existing interface (from `Prelude` and `System.Directory`).

In the end, we want to write a file. A file can be written by specifying either its relative path or absolute path. From that point of view, I think current polymorphic signatures make the most sense.
Still, I see why it would be nice to be explicit about whether the paths are relative or absolute.

To sum up, I don't know :smile:, but here are my preferences from favorite to least favorite:
1. Keep these functions polymorphic and see whether it causes problem. If it does, we revisit it. 
2. Add `readFileAbs` and `removeFileAbs` (which would just be differently-typed aliases for `readFile`) and use those.
3. Revert to explicit conversions and regular `readFile :: FilePath -> IO String`

Reply:
I kept the names, but changed the signatures and added a few comments.